### PR TITLE
Fix bug: editing person does not edit persons in modules

### PR DESCRIPTION
**Pull request by ekweirui**
Currently, editing a person creates a new person object, and this new person object is used to replace the current person object in the `UniquePersonList`.

However, this replacement does not occur in the modules' set of persons.

Thus, this pull request aims to:
* Make editing person occur in modules' set of persons as well

### CODE REVIEW

**Review by cheeheng — COMMENTED** [review]
LGTM. Perhaps you could add a few tests to ensure this bug does not occur again.

File: src/main/java/seedu/address/model/module/Module.java [hunk 1]
```diff
@@ -151,6 +151,27 @@ public boolean isSameModule(Module otherModule) {
...
+
+    /**
+     * Replaces the person {@code target} in the module's set of persons with {@code editedPerson}.
+     *
+     * @param target The person in the module's set to be replaced.
+     * @param editedPerson The person to replace {@code target} in the module's set.
+     */
+    public void setPerson(Person target, Person editedPerson) {
+        persons.remove(target);
+        persons.add(editedPerson);
```
Comment (cheeheng) [inline src/main/java/seedu/address/model/module/Module.java:172]: Should there be an assert to make sure that `target` always exists?

**Review by teoyuqi — APPROVED** [review]
Thanks for noticing the bug!

**Comment by teoyuqi** [comment]
@cheeheng @ekweirui
I agree with @cheeheng that we should indeed add test cases to test for this. I was looking through the current tests and it seems like the reason as to why this bug was not detected was because the test case uses a `typicalAddressBook` which does not have any modules in it:
image

Hence, the `Module-Person` association is never tested.

I believe the underlying cause is that `Module` and `Person` objects are still being treated as isolated entities in our tests. This is because the `typicalAddressBook` is built upon the `typicalModules` and `typicalPersons`, which were both created much earlier on when our modules and persons did not have an association between them.

A possible solution to this issue would hence be to redefine `typicalModules` as `Module` objects which have associations with  `Person` objects from `typicalPersons`, and to remove the ability for tests to be conducted on addressbooks with either only `Module` or `Person` objects.

**Comment by Tan-Jia-Rong** [comment]
Currently our implementation for `TypicalAddressBook` is as follows.

image

With this change modules and contacts are no longer a separate entity. 

Then, it is true that there exist a container-container relationship between modules and contacts.

That is contacts can exist on its own without modules.

Hence, I feel that we can retain the part on `getTypicalAddressBookWithOnlyPerson`. 

One solution would be to 'merge' `getTypicalAddressBook` and `getTypicalAddressBookWithModule` together into one function.

Alternatively, we could explore creating another function `getTypicalModuleWithAssociation` within TypicalModule and update `getTypicalAddressBook`. With this, we could retain all 3 funcitions currently in our TypicalAddressBook without much editing.

**Review by Tan-Jia-Rong — APPROVED** [review]
LGTM!

**Review by teoyuqi — APPROVED** [review]
LGTM

**Pull request merged**